examples/zero_copy/c++/zero_copy_subscriber.cxx

```diff
@@ -235,76 +245,75 @@ void ZeroCopyListener::on_data_available(DataReader* reader)
-
-                // Set the source timestamp as the time when the shared memory
-                // segment was updated
-                RTINtpTime_packFromNanosec(
-                        source_time,
-                        info_seq[i].source_timestamp.sec,
-                        info_seq[i].source_timestamp.nanosec);
-
-                // Compute the difference and store in current_time
-                latency_time = current_time;
-                RTINtpTime_decrement(latency_time, source_time);
-
-                // Keep track of total latency for computing averages
-                RTINtpTime_unpackToNanosec(
-                        latency.sec,
-                        latency.nanosec,
-                        latency_time);
-                total_latency = total_latency + latency;
-                received_frames++;
-
-                if (arg.verbose) {
-                    std::cout << "Received frame " << received_frames
-                            << " from DataWriter with key "
-                            << frame_set_view->key << " with checksum: "
-                            << frame->checksum << std::endl;
-                }
-
-                if (received_frames == 1) {
-                    previous_print_time = current_time;
-                }
-
-                if (((current_time.sec - previous_print_time.sec) >= 1)
-                        || (received_frames == arg.sample_count)) {
-                    previous_print_time = current_time;
-                    print_average_latency();
-                }
-
-            } else {
+            if (frame_set_view == NULL) {
                 std::cout << "Received data from an un recognized frame "
                         "writer" << std::endl;
+                continue;
             }
 
+            // Get the frame checksum from the sample
+            received_checksum = data_seq[i].checksum;
+
+            // Get the frame from the shared memory segment
+            // at the provided index
+            const Frame *frame = (*frame_set_view)[data_seq[i].index];
+
+            // Compute the checksum of the frame
+            // We only compute the checksum of the first 4 bytes. These
+            // are the bytes that we change in each frame for this example
+
+            // We dont compute CRC for the whole frame because it would add
+            // a latency component not due to the middleware. Real
+            // application may compute CRC differently
+            computed_checksum = RTIOsapiUtility_crc32(
+                    frame->get_buffer(),
+                    sizeof(int),
+                    0);
+
+            // Validate the computed checksum with the checksum sent by the
+            // DataWriter
+            if (computed_checksum != received_checksum) {
+                std::cout << "Checksum NOT OK" << std::endl;
+            }
+
+            // Get the current time for computing latency
+            get_current_time(clock, &current_time);
```

Comment:
I meant to encapsulate more than just getting the current time, also packing, computing the diff and unpacking. compute_latency?